### PR TITLE
docs: add daemon config file doc

### PR DESCRIPTION
**Pull request by Ace-Tang**
Signed-off-by: Ace-Tang <aceapril@126.com>

### CODE REVIEW

**Comment by allencloud** [comment]
markdown lint files, please write the document in markdownlint tool's rules. @Ace-Tang 
😄

File: docs/user/config_file.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+# Configuring pouch daemon config file
+
+We recommend users to set daemon flag through daemon config file, the default
+path to config file is `/etc/pouch/config.json`, you can change it by set
+value of `--config-file`.
+
+## Note
+
+1. The same flag(exclude slice or array type) can not be set through command
+line and config file 同时.
```
Comment (fuweid) [inline docs/user/config_file.md]: please remove the `同时` here. :)

File: docs/user/config_file.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+We recommend users to set daemon flag through daemon config file, the default
+path to config file is `/etc/pouch/config.json`, you can change it by set
+value of `--config-file`.
+
+### Note
+
+* The same flag(exclude slice or array type) can not be set through command
+line and config file simultaneously.
+* We allow users set slice or array type of flag simultaneously from command
+and config file line，and merge them.
```
Comment (Ace-Tang) [inline docs/user/config_file.md]:
```
#!/bin/bash -eo pipefail
find  ./ -name  "*.md" | grep -v vendor | grep -v extra |  grep -v commandline |  grep -v .github |  grep -v swagger |  grep -v api |  xargs mdl -r ~MD013,~MD024,~MD029,~MD033,~MD036
./docs/user/config_file.md:26: MD032 Lists should be surrounded by blank lines
./docs/user/config_file.md:28: MD032 Lists should be surrounded by blank lines
Exited with code 123
```
I do not see why this is wrong

File: docs/user/config_file.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+## Configuring pouch daemon config file
+
+We recommend users to set daemon flag through daemon config file, the default
+path to config file is `/etc/pouch/config.json`, you can change it by set
+value of `--config-file`.
+
+### Note
+
+* The same flag(exclude slice or array type) can not be set through command
```
Comment (HusterWan) [inline docs/user/config_file.md]: maybe need a blank line here

Comment (Ace-Tang) [reply docs/user/config_file.md]: no, if add a blank line here, it will cause another error,  which means just allow one blank line

Comment (Ace-Tang) [reply docs/user/config_file.md]: The rules confuse me very much !!!

Comment (fuweid) [reply docs/user/config_file.md]: Hi @Ace-Tang , the rule is here. https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md032---lists-should-be-surrounded-by-blank-lines

I think you can add the indent in the newline. Please have a try. haha

Comment (Ace-Tang) [reply docs/user/config_file.md]: As the text say
```
Some text

* Some
* List
```

I must add a text before `*some`, but if I do not want to add text, I just want `* some` follow `## title`, how do I need to do.

**Comment by allencloud** [comment]
Still wondering where to locate this document file. If we add a new folder user, I think we should update the docs/README.md .
How about adding all these into file pouchd.md in folder docs/commandline? @Ace-Tang

**Comment by Ace-Tang** [comment]
I do not think it proper to put the file under docs/commandline. @allencloud ， docs/commandline should include command line related info.

**Comment by HusterWan** [comment]
I will merge this pr first, if has any problem, feel free to contact me thanks, @Ace-Tang @allencloud @fuweid

**Pull request merged**